indico/modules/events/registration/util.py

```diff
@@ -412,7 +413,14 @@ def modify_registration(registration, data, management=False, notify_user=True):
                 personal_data_changes[key] = value
             setattr(registration, key, value)
     if not management and regform.needs_publish_consent:
-        registration.consent_to_publish = data.get('consent_to_publish', RegistrationVisibility.nobody)
+        consent_to_publish = data.get('consent_to_publish', RegistrationVisibility.nobody)
+        changes = make_diff_log({'consent_to_publish': (registration.consent_to_publish, consent_to_publish)},
+                                {'consent_to_publish': {'title': 'Consent to publish'}})
+        registration.consent_to_publish = consent_to_publish
+        registration.log(EventLogRealm.participants, LogKind.change, 'Registration',
+                         f'Consent to publish modified: {registration.full_name}',
```

Comment:
It looks like you are logging this now even if nothing was changed (ie someone just editing their registration w/o touching it)...